### ISSUE TITLE
REGEXP_MATCH problem: \b + accented characters does not match properly

### ISSUE THREAD

**Issue by GoogleCodeExporter** [runt18/google-bigquery#72]
```
What steps will reproduce the problem?

Run this query: 

Select Regexp_extract(StringToParse,r'\b(à)\b') as Extract,
 Regexp_match(StringToParse,r'\b(à)\b') as match,
FROM
(SELECT 'Voilà la séance qui est à Paris.' as StringToParse)


What is the expected output? What do you see instead?

I get:

    1   null    false   

I should get:

    1   à  true     

If I run: 

Select Regexp_extract(StringToParse,r'\b(séance)\b') as Extract,
 Regexp_match(StringToParse,r'\b(séance)\b') as match,
FROM
(SELECT 'Voilà la séance qui est à Paris.' as StringToParse)

I get: 

   1    séance true     


What version of the product are you using? On what operating system?

Not sure!

Please provide any additional information below.

Discussion here: 
http://stackoverflow.com/questions/22160406/bigquery-regexp-match-and-accents-bo
undary-wildcard-fails/22205809?noredirect=1#comment33713424_22205809
```

Original issue reported on code.google.com by `wgwin...@gmail.com` on 5 Mar 2014 at 7:56

**Comment by GoogleCodeExporter** [comment]
```
Forgot to mention: I am using the console (https://bigquery.cloud.google.com/) 
and apps scripts to run the queries that fail.
```

Original comment by `wgwin...@gmail.com` on 5 Mar 2014 at 7:59

**Comment by GoogleCodeExporter** [comment]
```
Hi - apologies for the delayed response, we're working on fixing the handling 
of multibyte characters; the fix should be live in the next week or two.  We'll 
update this issue when the fix is live.
```

Original comment by `thomasp...@google.com` on 4 Jun 2014 at 9:52
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Sorry for the long delay here. We thought this was a bug, but it turns out it's 
actually working as expected.

BigQuery uses RE2 for its regular expression parsing. RE2's syntax is 
documented here:

  https://code.google.com/p/re2/wiki/Syntax

According to that documentation, RE2's character classes are:

  \b = at word boundary (\w on one side and \W, \A, or \z on the other)
  \w = word characters (≡ [0-9A-Za-z_])
  \W = not word characters (≡ [^0-9A-Za-z_])
  \A = beginning of text
  \z = end of text

In other words, you can only use \b to match boundaries of non-accented 
characters. RE2 has plenty of support for Unicode characters, though, so you 
can most likely craft an alternative regexp using something like \pL.

I'll follow up on the linked StackOverflow question too.
```

Original comment by `jcon...@google.com` on 8 Jul 2014 at 4:24
- Changed state: **Invalid**
